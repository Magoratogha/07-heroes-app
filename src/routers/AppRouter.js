import { BrowserRouter, Route, Routes } from "react-router-dom";
import LoginScreen from "../components/login/LoginScreen";
import DashboardRoutes from "./DashboardRoutes";
import PrivateRoutes from "./PrivateRoutes";
import PublicRoutes from "./PublicRoutes";

function AppRouter() {
    return (
        <BrowserRouter>
            <Routes>
                <Route path="/login" element={
                    <PublicRoutes>
                        <LoginScreen />
                    </PublicRoutes>
                } /> 
                <Route path="/*" element={
                    <PrivateRoutes>
                        <DashboardRoutes />
                    </PrivateRoutes>
                } /> 
            </Routes>
        </BrowserRouter>
    )
}

export default AppRouter;