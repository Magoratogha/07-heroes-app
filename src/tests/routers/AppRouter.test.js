import { render, screen } from "@testing-library/react";
import { AuthContext } from "../../auth/authContext";
import AppRouter from "../../routers/AppRouter";

describe('Test on <AppRouter />', () => {
    
    const contextValue = {
        user: {
            logged: false
        }
    };

    it('should render LoginScreen if user is not authenticated', () => {
        const { container } = render(<AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>);
        expect(container).toMatchSnapshot();
        expect(screen.getByText('Login Screen', { selector: 'h1' })).toBeInTheDocument();
    });
});