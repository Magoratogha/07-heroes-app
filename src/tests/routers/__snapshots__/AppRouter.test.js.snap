// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> should render LoginScreen if user is not authenticated 1`] = `
<div>
  <div
    class="container mt-5"
  >
    <h1
      class="display-6"
    >
      Login Screen
    </h1>
    <hr />
    <button
      class="btn btn-outline-dark"
    >
      Login
    </button>
  </div>
</div>
`;
